### PR TITLE
Fixed #3906 - Pre-filled RELATE TO field in compose email pop up window isn't working for CASE in 7.9.x

### LINKED ISSUE #3906: Pre-filled RELATE TO field in compose email pop up window isn't working for CASE in 7.9.x
#### Issue

When you composing email in record in subpanel HISTORY in old email system 7.8.x was pre-filled the RELATE TO field by record name in case module.. now is it empty and must do this user and it consumes time :(

#### Expected Behavior

Will pre-filled RELATE TO field in compose window of email, when is email sending from case record

#### Actual Behavior

RELATE TO field is empty

#### Steps to Reproduce

1. Go to any case record
2. Click in History/Activity sub-panel button Compose Email
3. Look on RELATE to field in compose pop up window of email

#### Your Environment

* SuiteCRM Version used: SuiteCRM 7.9.2
* Browser name and version: Chrome 56.x 64bit
* Environment name and version: MySQL 5.6, PHP 7
* Operating System and version: Win 10 Home

### PR DESCRIPTION
**Pull request by Dillon-Brown**
Task: SCRM-658

## Description

When composing an email from "Cases"->"Detail-View"->"Activities-Sub-panel"->"Compose Email", the related to drop-down and field name do not populate.

## Motivation and Context

This change ensures that when a module like "Cases" does not have an email1 field that the composer will still relate the email.

Issue reference: #3906

## How To Test This

1. Go into "Cases"->"Detail-View".
2. Enter the sub-panel "Activities"->"Actions Drop-down"->"Compose Email".
3. Check that the email popup composer has a blank "To" field and that the email is related to the case.
4. Check that the same functionality is present in other modules like "Accounts" but with the "To" field populated.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

**Comment by Mausino** [comment]
@salesagility @samus-aran  the patch working on my local instance.. but is stack here from last year...  should somebody Resoleve Conflicts  ? that will merged

**Comment by samus-aran** [comment]
Thanks @Mausino 

@Dillon-Brown If you are available today to resolve conflicts that would be ideal!

File: include/generic/SugarWidgets/SugarWidgetSubPanelTopComposeEmailButton.php [hunk 1]
```diff
@@ -44,50 +44,68 @@
...
+     *
+     * @param bool $buttonSuffix If set to true the widget is a button
+     * @return string label of the widget's unique id
+     */
     public function getWidgetId($buttonSuffix = true)
     {
-    	global $app_strings;
-		$this->form_value = $app_strings['LBL_COMPOSE_EMAIL_BUTTON_LABEL'];
-    	return parent::getWidgetId();
+        global $app_strings;
```
Comment (gymad) [inline include/generic/SugarWidgets/SugarWidgetSubPanelTopComposeEmailButton.php]: Do not access globals, pass it as an (optional) dependency.

File: include/generic/SugarWidgets/SugarWidgetSubPanelTopComposeEmailButton.php [hunk 2]
```diff
@@ -44,50 +44,68 @@
...
-		$parent_id = $defines['focus']->id;
+    /**
+     * This function is used to create the HTML for a button.
+     *
+     * @param array $defines
+     * @param array $additionalFormFields
+     * @param bool $nonbutton
+     * @return string HTML
+     */
+    function display($defines, $additionalFormFields = null, $nonbutton = false)
```
Comment (gymad) [inline include/generic/SugarWidgets/SugarWidgetSubPanelTopComposeEmailButton.php]: - function visibility is `public`.
- function complexity is to large.

File: include/generic/SugarWidgets/SugarWidgetSubPanelTopComposeEmailButton.php [hunk 3]
```diff
@@ -44,50 +44,68 @@
...
+            $client = $userPref;
+        } else {
+            $client = $defaultPref;
+        }
+        if ($client != 'sugar') {
+            $bean = $defines['focus'];
+            // awu: Not all beans have emailAddress property, we must account for this
+            if (isset($bean->emailAddress)) {
+                $to_addrs = $bean->emailAddress->getPrimaryAddress($bean);
+                $button = "<input class='button' type='button'  value='$value'  id='" . $this->getWidgetId() . "'  name='" . preg_replace('[ ]',
```
Comment (gymad) [inline include/generic/SugarWidgets/SugarWidgetSubPanelTopComposeEmailButton.php]: Try to keep html code separated, use templates.

File: modules/Emails/EmailUI.php [hunk 4]
```diff
@@ -5,7 +5,7 @@
  * SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
  *
  * SuiteCRM is an extension to SugarCRM Community Edition developed by SalesAgility Ltd.
- * Copyright (C) 2011 - 2016 SalesAgility Ltd.
+ * Copyright (C) 2011 - 2017 SalesAgility Ltd.
```
Comment (gymad) [inline modules/Emails/EmailUI.php]: 2017.. please update the licence.

**Comment by Dillon-Brown** [comment]
Issue appears to be resolved

**Pull request closed without merging**